src/VisualStudio/Core/Def/Implementation/ProjectSystem/FileChangeTracker.cs

```diff
@@ -65,44 +83,81 @@ public void StartFileChangeListeningAsync()
-                Marshal.ThrowExceptionForHR(
-                    _fileChangeService.AdviseFileChange(_filePath, FileChangeFlags, this, out var newCookie));
-                return newCookie;
+                try
+                {
+                    Marshal.ThrowExceptionForHR(
+                        _fileChangeService.AdviseFileChange(_filePath, FileChangeFlags, this, out var newCookie));
+                    return newCookie;
+                }
+                catch (Exception e) when (ShouldTrapException(e))
+                {
+                    return null;
+                }
             }, LazyThreadSafetyMode.ExecutionAndPublication);
 
-            // file change service is free-threaded. start running it in background right away
-            Task.Run(() => _fileChangeCookie.Value, CancellationToken.None);
+            lock (s_lastBackgroundTaskGate)
+            {
+                s_lastBackgroundTask = s_lastBackgroundTask.ContinueWith(_ => _fileChangeCookie.Value, CancellationToken.None, TaskContinuationOptions.None, TaskScheduler.Default);
+            }
+        }
+
+        private static bool ShouldTrapException(Exception e)
+        {
+            if (e is FileNotFoundException)
+            {
+                // The IVsFileChange implementation shouldn't ever be throwing exceptions like this, but it's a
+                // transient file system issue (perhaps the file being deleted while we're changing subscriptions)
+                // and so there's nothing better to do. We'll still non-fatal to track the rate this is happening
+                return FatalError.ReportWithoutCrash(e);
+            }
+            else if (e is PathTooLongException)
+            {
+                // Nothing better we can do. We won't be able to open this file either, and thus we'll do our usual
+                // reporting of unopenable/missing files to the output window as usual.
+                return true;
+            }
+            else
+            {
+                return false;
+            }
         }
 
         public void StopFileChangeListening()
         {
             if (_disposed)
             {
-                throw new ObjectDisposedException(typeof(FileChangeTracker).Name);
+                throw new ObjectDisposedException(nameof(FileChangeTracker));
             }
 
             // there is a slight chance that we haven't subscribed to the service yet so we subscribe and unsubscribe
             // both here unnecessarily. but I believe that probably is a theoretical problem and never happen in real life.
             // and even if that happens, it will be just a perf hit
-            if (_fileChangeCookie != s_none)
+            if (_fileChangeCookie == s_none)
             {
-                var hr = _fileChangeService.UnadviseFileChange(_fileChangeCookie.Value);
+                return;
+            }
+
+            var fileChangeCookie = _fileChangeCookie.Value;
+            _fileChangeCookie = s_none;
```

Comment:
This type being used across threads (other than with EnsureSubscription) has never been safe, really...